agent/command.go

```diff
@@ -147,6 +147,11 @@ func (a *Agent) runCommandSet(ctx context.Context, tc *taskContext, commandInfo
 			return errors.Wrap(ctx.Err(), "Agent stopped early")
 		}
 		tc.logger.Task().Infof("Finished %s in %s", fullCommandName, time.Since(start).String())
+		if (options.isTaskCommands || options.failPreAndPost) && a.endTaskResp != nil && !a.endTaskResp.ShouldContinue {
+			// only error if we're running a command that should fail, and we don't want to continue to run other tasks
+			tc.logger.Task().Debug("task status has been set; triggering end task")
```

Comment:
Although it would be a duplicate, this log line should log `endTaskResp.Status` I think, in case we somehow don't get to the validation log line. This allows a user to see earlier what status they've set and would aid in debugging.